lib/ace/ext/papeeria/latex_behaviour.coffee

```diff
@@ -13,77 +32,89 @@ define((require, exports, module) ->
     lastState = if typeof(state) == "string" then state else state[state.length - 1]
     lastPrevState = if typeof(pState) == "string" then pState else pState[pState.length - 1]
     return (
-      (not token? and /math/.test(lastPrevState)) or
-      (token? and /equation/.test(token.type)) or
-      (nextToken? and /equation/.test(nextToken.type)) or
-      (not nextToken? and /math/.test(lastState))
+        # This handles the case, when the cursor is on the empty string
+        (not token? and lastPrevState.indexOf(MATH_TYPE) > -1) or
+        # This handles the common case, when the cursor is inside the equation
+        (token? and token.type.indexOf(EQUATION_TYPE) > -1) or
+        # This handles the specific case, when the cursor is on the very start of the equation
+        (nextToken? and nextToken.type.indexOf(EQUATION_TYPE) > -1)
     )
 
+
+  DO_NOTHING = null
```

Comment:
Why create a new object each time? @dbarashev actually told me to carry out these definitions. And about moving it higher -- don't see much of a point to move declarations further from where they are used.